### PR TITLE
Analysis Exports: Multiple Selection UI

### PR DESCRIPTION
**Pull request by ChrisHuynh333**
## What does this PR do and why?
This PR adds the UI to select multiple workflow executions for export.
Main changes include:
- Checkboxes and select/deselect all feature for workflow execution tables
- Fix data export preview tab for analysis exports to include multiple workflow executions
- Refactor SampleActions (for listing) to ListActions to be inclusive of listing both samples and workflow executions

## Screenshots or screen recordings
[Screencast from 2024-08-27 02:39:22 PM.webm](https://github.com/user-attachments/assets/ef2302cc-8442-483f-b314-ed1b22355d84)

## How to set up and validate locally
1. Set up the ability to run workflow executions locally as described in [PR 507](https://github.com/phac-nml/irida-next/pull/507)
2. Run a couple workflow executions
3. Navigate to the user workflow executions page and test the select/deselect all buttons and ensure they behave as expected.
4. Select multiple workflow executions and create an export
5. In the dialog, verify that the selection number and listed workflow execution data is accurate.
6. Submit the export
7. After 30s, download the export and verify the export contents are as expected.
8. Click the `Preview` tab and ensure the page listing is as expected.
9. Repeat the above with workflow executions from `Automated Workflow Executions`, The workflows can be set up with the steps described in [PR585](https://github.com/phac-nml/irida-next/pull/585). 
10. Create an export from a workflow execution's `show` page, and verify steps 5-8.

Other tests:
- Verify that the other data export, transfer samples, and clone samples dialogs still list the samples appropriately as the logic for this was refactored

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.

- [x] I have evaluated the [PR acceptance checklist](https://phac-nml.github.io/irida-next/docs/development/development_processes/code_review#acceptance-checklist) for this PR.

### CODE REVIEW

File: app/controllers/concerns/list_actions.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+# Common sample actions
```
Comment (ksierks) [inline app/controllers/concerns/list_actions.rb:3]: Comment is out of date.

File: test/controllers/data_exports_controller_test.rb [hunk 2]
```diff
@@ -275,19 +275,20 @@ class DataExportsControllerTest < ActionDispatch::IntegrationTest
...
+    get redirect_data_export_path(@data_export1, identifier: 'INXT_SAM_AAAAAAAAAA')
     assert_response :redirect
   end
 
   test 'should list samples' do
     post list_data_exports_path(format: :turbo_stream), params: {
       page: 1,
-      sample_ids: [@sample1.id]
+      sample_ids: [@sample1.id],
+      list_class: 'sample'
```
Comment (ksierks) [inline test/controllers/data_exports_controller_test.rb:291]: Is it worth testing the workflow_execution list_class get tested too?

**Comment by ksierks** [comment]
This works great! My only thought is, does it makes sense to disable the checkbox for workflow executions that are not in a completed state? Thanks!

**Comment by ChrisHuynh333** [comment]
> This works great! My only thought is, does it makes sense to disable the checkbox for workflow executions that are not in a completed state? Thanks!

I played around with this while working this PR up because I had the same thought. It gets a little complicated as disabling checkboxes won't disable the ability to select them via the select all buttons. And then even if we added the select all to filter only `completed` WEs, we'd likely want to add a delete multiple functionality which would need to include additional WE states.

I asked in standup this week about having an included/excluded listing for the user in the create modal and then filter out on the backend in the job, as I think that will probably be the best way to move forward with this in the future, and it was decided we'll leave this as is for now and perhaps revisit it in the future based on user feedback.

File: app/controllers/concerns/list_actions.rb (hunk 1 above)
Comment (ChrisHuynh333) [reply app/controllers/concerns/list_actions.rb]: Fixed in f781876c5c17b14660db3619e035a4ff5bb3d3ec

File: test/controllers/data_exports_controller_test.rb (hunk 2 above)
Comment (ChrisHuynh333) [reply test/controllers/data_exports_controller_test.rb:291]: Added in f781876c5c17b14660db3619e035a4ff5bb3d3ec

**Review by ksierks — APPROVED** [review]
Looks good to me!

**Review by joshsadam — APPROVED** [review]
Great work, this appears to all work as advertised.

**Pull request merged**